tracer/src/Datadog.Trace/Iast/IastRequestContext.cs

```diff
@@ -36,16 +36,23 @@ internal static void AddIastDisabledFlagToSpan(Span span)
 
     internal void AddIastVulnerabilitiesToSpan(Span span)
     {
-        span.Tags.SetTag(Tags.IastEnabled, "1");
-
-        if (_vulnerabilityBatch != null)
+        try
         {
-            span.Tags.SetTag(Tags.IastJson, _vulnerabilityBatch.ToString());
-        }
+            span.Tags.SetTag(Tags.IastEnabled, "1");
+
+            if (_vulnerabilityBatch != null)
+            {
+                span.Tags.SetTag(Tags.IastJson, _vulnerabilityBatch.ToString());
+            }
 
-        if (_executedTelemetryHelper != null)
+            if (_executedTelemetryHelper != null)
+            {
+                _executedTelemetryHelper.GenerateMetricTags(span.Tags, _taintedObjects.GetEstimatedSize());
+            }
+        }
+        catch
         {
-            _executedTelemetryHelper.GenerateMetricTags(span.Tags, _taintedObjects.GetEstimatedSize());
+            Log.Error("Error in IAST AddIastVulnerabilitiesToSpan");
```

Comment:
Don't you want to log the exception here? 🤔

Reply:
Sure! Thanks!